apps/explorer/src/pages/transaction-result/SendReceiveView.tsx

```diff
@@ -1,56 +1,243 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
+
+import { COIN_TYPE_ARG_REGEX } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
 import cl from 'clsx';
+import { useState, useContext } from 'react';
 
+import { ReactComponent as DoneIcon } from '../../assets/SVGIcons/16px/CheckFill.svg';
+import { ReactComponent as StartIcon } from '../../assets/SVGIcons/Start.svg';
 import Longtext from '../../components/longtext/Longtext';
+import { NetworkContext } from '../../context';
+import { DefaultRpcClient as rpc } from '../../utils/api/DefaultRpcClient';
+import { parseObjectType } from '../../utils/objectUtils';
 
 import styles from './SendReceiveView.module.css';
 
+import { useFormatCoin, CoinFormat } from '~/hooks/useFormatCoin';
+import { Heading } from '~/ui/Heading';
+
 type TxAddress = {
-    timestamp_ms?: number;
     sender: string;
     recipient?: string[];
+    amount?: bigint[];
+    objects?: string[];
 };
-//TODO: Add date format function
-function SendRecieveView({ data }: { data: TxAddress }) {
+
+const getObjType = (objId: string, network: string) =>
+    rpc(network)
+        .getObject(objId)
+        .then((obj) => parseObjectType(obj));
+
+const getCoinLabelFromObjType = (objType: string) =>
+    COIN_TYPE_ARG_REGEX.exec(objType)?.[1];
+
+function MultipleRecipients({ sender, recipient, amount, objects }: TxAddress) {
+    const [network] = useContext(NetworkContext);
+
+    const [isSingleCoin, setIsSingleCoin] = useState(false);
```

Comment:
This shouldn't be state, it should just be derived from the `data` of the useQuery.